sysbuild/Kconfig.zip

```diff
@@ -44,6 +44,46 @@ config DFU_ZIP_BLUETOOTH_MESH_METADATA
 	help
 	  Includes metadata for bluetooth mesh.
 
+if DFU_ZIP_BLUETOOTH_MESH_METADATA
+
+choice DFU_ZIP_BLUETOOTH_MESH_METADATA_FWID
+	prompt "Source of Firmware ID to include in metadata file"
+	default DFU_ZIP_BLUETOOTH_MESH_METADATA_FWID_CUSTOM
+
+config DFU_ZIP_BLUETOOTH_MESH_METADATA_FWID_CUSTOM
+	bool "Use a custom Firmware ID"
+	help
+	  The value to include as Firmware ID in the Bluetooth Mesh metadata file is set using
+	  ``SB_CONFIG_DFU_ZIP_BLUETOOTH_MESH_METADATA_FWID_CUSTOM_HEX``.
+	  Note: You are required to make sure that the Firmware ID used when initializing the DFU
+	  Server matches the ID in the metadata file.
+
+config DFU_ZIP_BLUETOOTH_MESH_METADATA_FWID_MCUBOOT_VERSION
+	bool "Use a Firmware ID generated from the MCUboot version"
+	help
+	  Includes a Firmware ID in the Bluetooth Mesh metadata file consisting of the values of
+	  ``CONFIG_BT_COMPANY_ID`` and ``CONFIG_MCUBOOT_IMGTOOL_SIGN_VERSION``.
+	  Note: You are required to make sure that the Firmware ID used when initializing the DFU
+	  Server matches the ID in the metadata file.
+
+endchoice
+
+if DFU_ZIP_BLUETOOTH_MESH_METADATA_FWID_CUSTOM
+
+config DFU_ZIP_BLUETOOTH_MESH_METADATA_FWID_CUSTOM_HEX
+	string "HEX string to use as Firmware ID"
+	help
+	  The raw value of this option is included as the Firmware ID in the Bluetooth Mesh metadata
+	  file. This must be a valid hex string, and must at minimum begin with a Company ID in
+	  little-endian order. The rest of the string is vendor-specific version information and may
+	  be empty.
```

Comment:
(copied from https://github.com/nrfconnect/sdk-nrf/pull/20255#discussion_r1949023996)

I'm still uncertain about this. The model spec (I reffered initially to Access layer, but model layer has the same requirement) requires all data to be encoded into little endianness (section 3.8.1). The DFU server in our stack simply picks what points to fwid and passes it over the mesh network:
https://github.com/zephyrproject-rtos/zephyr/blob/baced6d315f0c58cf4950a8c49cb7a49d18c208f/subsys/bluetooth/mesh/dfu_srv.c#L171-L172

This works fine with our stack because we pass the fwid "as is" (as it comes in the message). But if some other stack is used, it can convert fwid into system endianness, and we will have problems here.

Reply:
It cannot convert FWID into system endianness, because it doesn't know what it contains.

If I get the bytes "590001020304" and don't know the vendor-specific scheme, the only thing I know is that "5900" is a 16 bit little endian company ID (0x0059). But is the remaining part a 32-bit little endian ID (0x04030201) or 4 different 8-bit numbers {0x01, 0x02, 0x03, 0x04}? No way to tell. No way to tell the endianness or to swap it, unless you know the vendor-specific encoding.

Reply:
Ah, yeah, seems correct

Reply:
From the spec/IOP perspective, the FWID should be considered as 
`{ uint16_t cid; uint8_t[] vendor_specific }` if you ask me, and a byte array has no endianness.